lib/manager/terraform/lockfile/util.ts

```diff
@@ -0,0 +1,204 @@
+import { readLocalFile } from '../../../util/fs';
+import { get as getVersioning } from '../../../versioning';
+import type { UpdateArtifactsResult } from '../../types';
+import type {
+  LineNumbers,
+  ProviderLock,
+  ProviderLockUpdate,
+  ProviderSlice,
+} from './types';
+
+const providerStartLineRegex = /^provider "(?<registryUrl>[^/]*)\/(?<namespace>[^/]*)\/(?<depName>[^/]*)"/;
+const versionLineRegex = /^(?<prefix>[\s]*version[\s]*=[\s]*")(?<version>[^"']+)(?<suffix>".*)$/;
+const constraintLineRegex = /^(?<prefix>[\s]*constraints[\s]*=[\s]*")(?<constraint>[^"']+)(?<suffix>".*)$/;
+const hashLineRegex = /^(?<prefix>\s*")(?<hash>[^"]+)(?<suffix>",.*)$/;
+
+const lockFile = '.terraform.lock.hcl';
```

Comment:
So for now only repo root lock files are supported?

Reply:
Yes, but this should easy to add later on.